vars/asfMainNetBeansBuild.groovy

```diff
@@ -371,13 +355,26 @@ def doParallelClusters(cconfigs) {
                                     def timestamp = sh(returnStdout: true, script: 'date +%y%m%d').trim()
 
                                     sh "cd distpreparation${versionnedpath}installer && ./installer.sh ${binaryfile} ${version} ${timestamp}"
-                                    sh "cp distpreparation${versionnedpath}installer/dist/bundles/* dist/installers/ "
-                                    publishToNightlies("/netbeans/candidate/installers","distpreparation${versionnedpath}installer/dist/bundles/*","distpreparation${versionnedpath}installer/dist/bundles/")
+                                    // we archive put to nightlies only exe for window, nbpackage is intended to do the installler
+                                    sh "cp distpreparation${versionnedpath}installer/dist/bundles/*.exe dist/installers/ "
+                                    publishToNightlies("/netbeans/candidate/installers","distpreparation${versionnedpath}installer/dist/bundles/*.exe","distpreparation${versionnedpath}installer/dist/bundles/")
                              
                                     sh "rm -rf distpreparation${versionnedpath}installer/dist"
                                     // XXX take too long 18012023 publishToNightlies("/netbeans/candidate/installerspreparation","distpreparation/**/**","distpreparation")
                                                                   
                                     archiveArtifacts 'distpreparation/**'  
+                                    
+                                    sh "mkdir -p nbpackage${versionnedpath}installer"
+                                    withMaven(maven:tooling.myMaven,jdk:tooling.jdktool,publisherStrategy: 'EXPLICIT',mavenLocalRepo: localRepo,options:[artifactsPublisher(disabled: true)])
+                                    {
+                                        // unpack nbpackage snapshot can later
+                                        sh "mvn org.apache.maven.plugins:maven-dependency-plugin:3.5.0:unpack -DoutputDirectory=nbpackage${versionnedpath}installer -Dartifact=org.apache.netbeans:nbpackage:${nbpackageversion}:bin:zip -Dmaven.repo.local=${env.WORKSPACE}/.repository -DremoteRepositories=apache.snapshots.https::::https://repository.apache.org/snapshots"
+                                        
+                                        // build installer only deb for testing.
+                                        sh "nbpackage${versionnedpath}installer/nbpackage-${nbpackageversion}/bin/nbpackage --type linux-deb -Pname=\"Apache NetBeans\" -Pversion=${version} -Purl=\"https://netbeans.apache.org\" -Pdeb.maintainer=\"NetBeans Mailing List <users@netbeans.apache.org>\" -Pdeb.desktop-filename=\"apache-netbeans-ide-${version}\" -Pdeb.wmclass=\"Apache NetBeans IDE ${version}\" --input dist${versionnedpath}${path}-${rmversion}-bin.zip"
```

Comment:
`deb.desktop-filename` and `deb.wmclass` should probably be using `${rmversion}` here.  They need to include any suffix in the main window title.  `-Pversion` would be good as `${rmversion}` too probably, although using a hyphen breaks the RPM packager currently (hence 17~rc2 on nightlies).

Reply:
put in multiline string as it's unreadble.

groovy seems ok in my local jenkins but cannot go far